### ISSUE TITLE
canplay + currentTime

### ISSUE THREAD

**Issue by baw23** [sampotts/plyr#1848]
```
jQuery(document).ready(() => {
  
   jQuery('.player-src').on('click', function () {
      
      src = jQuery(this).data("src");
      playerTime = player.currentTime;
 
      player.source = {
        type: 'audio',
        title: 'Example title',
        sources: [
         {
              src: src,
              type: 'audio/mp3',
          }
        ],
        
      };
     
     player.once('canplay', event => {
       
   player.currentTime = playerTime;
       player.play();
});
     
     
   });
})
```

### Expected behaviour
When a button is clicked, current time is stored in a variable, the source is changed, current time is then recalled and the player starts again.

### Actual behaviour
This works perfectly - once. Using player.once means that it will happen on the first time the button is clicked, but does not work on subsequent clicks. Using player.on creates a loop.

**Comment by akuma06** [comment]
The canplay event is a HTML5 event that tells you when there is enough data buffered to play your media element (here audio). So it triggers when you change the source but also when you move into the file. So this is an expected behavior and not a bug per se. You should probably move your `player.currentTime = playerTime` outside the event since the source change is synchronous.

**Comment by baw23** [comment]
It says in another issue that currentTime only works for HTML5 players if placed in a canplay event. I tried moving it outside of the event, and yeah, doesn't work.

**Comment by akuma06** [comment]
Sorry I didn't read clearly what you wrote. I thought you were complaining that "on" was a bug since it would repeatedly call itself.
I've since fiddled a bit with the library and I found other bugs related to this and where it might be born.
First, your code is correct as it is. The "once" method is called once when you click the button but it is also set everytime you click it. So your listener is called "once" every time you click your button. Try to put a console.log, and you will see what I mean. A workaround is to put player.currentTime and player.play inside a timeout (200 preferably) (look at the link I provided).

Second, the bug isn't limited to only the canplay event. When you set currentTime, the player check if the duration is available, if it's not, it doesn't set the currentTime (https://github.com/sampotts/plyr/blob/ef85cb6195dc07627bd4fd357723c6ee96033ed9/src/js/plyr.js#L466).
However, when a video or audio html tag emit a "loadedmetadata", "loadeddata" and "canplay" event, the duration should already be set since the souce's metadata are loaded. If you debug the script at player.currentTime and ask for the duration during one of the event above, this will display a duration of 0 (NaN on the media.duration).
Furthermore, after more fiddling, it seems that those three events are called twice after a source change. First time the duration is equal to 0 but the source is correctly set (which is weird), second time the duration is finally correct.
Here is a sample code showing the debug output of Plyr Bug. (https://jsbin.com/libozac/edit?html,js,console,output).
Here is a sample code showing the HTML5 video events without Plyr: (https://jsbin.com/cukusow/edit?html,js,console,output)

I've looked through the code and I don't see a clear reason why those three events are called twice. From what I can see this happens only to the HTML5 plaform. I have a possibility worth looking into:
When a source change occurs, the media element is destroyed, another video element is created with sources but before that there is a blank video that is loading to cancel the previous request. I think the browser is still emitting the loadedmetadata event for this blank video. This would explain the weird duration = NaN.

**Comment by jayrajzala** [comment]
I did a similar thing using variables. set two global variables,
var oldTime = 0;
var allowSeeking = false;

`jQuery(document).ready(() => {
  
   jQuery('.player-src').on('click', function () {
      
      src = jQuery(this).data("src");
      oldTime = player.currentTime;
      allowSeeking = true;
 
      player.source = {
        type: 'audio',
        title: 'Example title',
        sources: [
         {
              src: src,
              type: 'audio/mp3',
          }
        ]};
     
     });

})

player.on('playing', function(){
if(allowSeeking === true){
player.currentTime = oldTime; 
allowSeeking=false;
}
})
`

**Comment by jayrajzala** [comment]
- I did a similar thing using variables. set a global variable for storing time,

`
var oldTime = 0;

jQuery(document).ready(() => {

jQuery('.player-src').on('click', function () {

  src = jQuery(this).data("src");
  oldTime = player.currentTime;
  player.source = {
    type: 'audio',
    title: 'Example title',
    sources: [
     {
          src: src,
          type: 'audio/mp3',
      }
    ]};

 });
})

player.on('loadedmetadata', function(){
if(oldTime !== 0){
player.play();
player.currentTime = oldTime;
}
})
`

**Comment by MacMaru** [comment]
It's the same for Vimeo sources: trouble restoring a saved `currentTime` upon load/ready. 
Works when duration is present in an `on('play')` handler, but I'd like to restore/seek to the last saved time before starting the video.
